sybilMTP.go

```diff
@@ -0,0 +1,213 @@
+package circuits
+
+import (
+	"encoding/json"
+	"fmt"
+	core "github.com/iden3/go-iden3-core"
+	"github.com/iden3/go-merkletree-sql/v2"
+	"github.com/pkg/errors"
+	"math/big"
+)
+
+type SybilMTPInputs struct {
+	BaseConfig
+
+	ID                       *core.ID `json:"id"`
+	ProfileNonce             *big.Int `json:"profileNonce"`
+	ClaimSubjectProfileNonce *big.Int `json:"claimSubjectProfileNonce"`
+
+	IssuerClaim ClaimWithMTPProof `json:"issuerClaim"`
+	HolderClaim ClaimWithMTPProof `json:"holderClaim"`
+
+	GISTProof GISTProof `json:"gistProof"`
+	CRS       string    `json:"crs"`
+
+	RequestID        string `json:"requestID"`
+	IssuerID         string `json:"issuerID"`
+	CurrentTimestamp string `json:"currentTimestamp"`
+}
+
+type sybilMTPCircuitInputs struct {
+	IssuerClaim           *core.Claim        `json:"issuerClaim"`
+	IssuerClaimMtp        []*merkletree.Hash `json:"issuerClaimMtp"`
+	IssuerClaimClaimsRoot *merkletree.Hash   `json:"issuerClaimClaimsRoot"`
+	IssuerClaimRevRoot    *merkletree.Hash   `json:"issuerClaimRevRoot"`
+	IssuerClaimRootsRoot  *merkletree.Hash   `json:"issuerClaimRootsRoot"`
+	IssuerClaimIdenState  *merkletree.Hash   `json:"IssuerClaimIdenState"`
+
+	IssuerClaimNonRevMtp      []*merkletree.Hash `json:"issuerClaimNonRevMtp"`
+	IssuerClaimNonRevMtpNoAux string             `json:"issuerClaimNonRevMtpNoAux"`
+	IssuerClaimNonRevMtpAuxHi *merkletree.Hash   `json:"issuerClaimNonRevMtpAuxHi"`
+	IssuerClaimNonRevMtpAuxHv *merkletree.Hash   `json:"issuerClaimNonRevMtpAuxHv"`
+
+	IssuerClaimNonRevClaimsRoot *merkletree.Hash `json:"issuerClaimNonRevClaimsRoot"`
+	IssuerClaimNonRevRevRoot    *merkletree.Hash `json:"issuerClaimNonRevRevRoot"`
+	IssuerClaimNonRevRootsRoot  *merkletree.Hash `json:"issuerClaimNonRevRootsRoot"`
+	IssuerClaimNonRevState      *merkletree.Hash `json:"IssuerClaimNonRevState"`
+
+	HolderClaim           *core.Claim        `json:"holderClaim"`
+	HolderClaimMtp        []*merkletree.Hash `json:"holderClaimMtp"`
+	HolderClaimClaimsRoot *merkletree.Hash   `json:"holderClaimClaimsRoot"`
+	HolderClaimRevRoot    *merkletree.Hash   `json:"holderClaimRevRoot"`
+	HolderClaimRootsRoot  *merkletree.Hash   `json:"holderClaimRootsRoot"`
+	HolderClaimIdenState  *merkletree.Hash   `json:"holderClaimIdenState"`
+
+	GistRoot     *merkletree.Hash   `json:"gistRoot"`
+	GistMtp      []*merkletree.Hash `json:"gistMtp"`
+	GistMtpAuxHi *merkletree.Hash   `json:"gistMtpAuxHi"`
+	GistMtpAuxHv *merkletree.Hash   `json:"gistMtpAuxHv"`
+	GistMtpNoAux string             `json:"gistMtpNoAux"`
+
+	CRS string `json:"crs"`
```

Comment:
`CRS` can not be a string. We need to discuss it.

Reply:
Updated